app/src/main/java/ch/epfl/sdp/drone3d/map/MapboxAreaBuilderDrawer.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2021  Drone3D-Team
+ * The license can be found in LICENSE at root of the repository
+ */
+
+/*
+ * This class was taken from the project Fly2Find and adapted for our project
+ */
+package ch.epfl.sdp.drone3d.map
+
+import android.graphics.Color
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.maps.MapView
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.plugins.annotation.*
+import com.mapbox.mapboxsdk.utils.ColorUtils
+
+/**
+ * Drawer on MapBox for an AreaBuilder
+ */
+class MapboxAreaBuilderDrawer(
+    mapView: MapView,
+    mapboxMap: MapboxMap,
+    style: Style,
+    onLongClickConsumed: () -> Boolean
+) : MapboxDrawer {
+
+
+    companion object {
+        private const val REGION_FILL_OPACITY: Float = 0.5F
+        private const val WAYPOINT_RADIUS: Float = 8f
+    }
+
+    val onVertexMoved = mutableListOf<(old: LatLng, new: LatLng) -> Unit>()
+
+    private var fillManager: FillManager = FillManager(mapView, mapboxMap, style)
+    private var circleManager: CircleManager = CircleManager(mapView, mapboxMap, style)
+
+    private lateinit var fillArea: Fill
+
+    private var reset: Boolean = false
+
+    private var nbVertices = 0
+
+    private val dragListener = object : OnCircleDragListener {
+        lateinit var previousLocation: LatLng
+        override fun onAnnotationDragStarted(annotation: Circle) {
+            previousLocation = annotation.latLng
+            onLongClickConsumed()
+        }
+
+        override fun onAnnotationDrag(annotation: Circle) {
+            onVertexMoved.forEach { it(previousLocation, annotation.latLng) }
+            previousLocation = annotation.latLng
+        }
+
+        override fun onAnnotationDragFinished(annotation: Circle?) {
+        }
+    }
+
+    init {
+        circleManager.addDragListener(dragListener)
+        circleManager.addLongClickListener { onLongClickConsumed() }
+    }
+
+    fun getUpperLayer(): String {
+        return circleManager.layerId
+    }
+
+    override fun onDestroy() {
+        onVertexMoved.clear()
+        nbVertices = 0
+        fillManager.deleteAll()
+        circleManager.deleteAll()
+        fillManager.onDestroy()
+        circleManager.onDestroy()
+    }
+
+    fun draw(pa: DrawableArea) {
```

Comment:
Maybe add some documentation for this function and change the name of the pa variable to make it easier to understand

Reply:
Done